packages/cli-hooks/src/get-hooks.js

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env node
+
+import { fileURLToPath } from 'url';
+import fs from 'fs';
+
+/**
+ * Implementation the get-hooks script hook required by the Slack CLI.
+ * Printed as an object containing featured provided by the SDK.
+ */
+
+if (fs.realpathSync(process.argv[1]) === fileURLToPath(import.meta.url)) {
+  console.log(JSON.stringify(getHooks())); // eslint-disable-line no-console
+}
+
+/**
+ * Standardized communication format between the SDK and CLI regarding hooks.
+ * @typedef SDKInterface
+ * @property {Record<string, string>} hooks - Commands available in the package.
+ * @property {SDKConfig} config - Settings for SDK and CLI communication.
+ * @property {string} runtime - Target runtime that app functions execute in.
+ */
+
+/**
+ * Additional configurations provided to the CLI about the SDK.
+ * @typedef SDKConfig
+ * @property {string[]} protocol-version - Named CLI protocols used by the SDK.
+ * @property {SDKConfigWatch} [watch] - Settings for file watching features.
+ * @property {boolean} [sdk-managed-connection-enabled] -
+ *   If the SDK or CLI manages websocket connections for run command executions.
+ */
+
+/**
+ * Information about the files to watch for specific changes.
+ * @typedef SDKConfigWatch
+ * @property {string} filter-regex - Regex pattern for finding filtered files.
+ * @property {string[]} paths - Specific locations to begin searching for files.
+ */
+
+/**
+ * Contains available hooks and other configurations available to the SDK.
+ * @returns {SDKInterface} Information about the hooks currently supported.
+ */
+export default function getHooks() {
+  return {
+    hooks: {
+      'get-manifest': 'npx -q --no-install -p @slack/cli-hooks slack-cli-get-manifest',
+      'check-update': 'npx -q --no-install -p @slack/cli-hooks slack-cli-check-update',
+      start: 'npx -q --no-install -p @slack/cli-hooks slack-cli-start',
+    },
+    config: {
+      watch: {
+        'filter-regex': '^manifest\\.json$',
+        paths: [
+          '.',
+        ],
+      },
+      'protocol-version': ['message-boundaries', 'default'],
```

Comment:
Should we pull these protocol names from the protocol source file in this package? The reason I bring this up is that `protocol.js` defines a `SUPPORTED_NAMED_PROTOCOLS` array that _only_ includes the message boundary protocol, but by returning this line in get-hooks, we're telling the CLI that either of the two named protocols is fine to use (with a preference for message-boundaries). We should decide on one approach.

Reply:
Totally in favor of using `SUPPORTED_NAMED_PROTOCOLS` - good call. Making the update now to use that exported array with only `message-boundaries` for now. Open to also including `default` in the export, but I lean towards including just the `message-boundaries` in this hook.